agreement/proposalTracker.go

```diff
@@ -144,17 +163,18 @@ func (t *proposalTracker) handle(r routerHandle, p player, e event) event {
 		}
 		t.Duplicate[v.R.Sender] = true
 
-		if t.Staging != bottom {
-			err := errProposalTrackerStaged{}
-			return filteredEvent{T: voteFiltered, Err: makeSerErr(err)}
-		}
-
-		var err error
-		t.Freezer, err = t.Freezer.accept(v)
+		newFreezer, effect, err := t.Freezer.accept(v)
+		t.Freezer.copyCredentialTrackingState(newFreezer)
 		if err != nil {
 			err := errProposalTrackerPS{Sub: err}
+			return filteredEvent{T: voteFiltered, CredentialTrackingNote: effect, Err: makeSerErr(err)}
+		}
+
+		if t.Staging != bottom {
+			err = errProposalTrackerStaged{}
 			return filteredEvent{T: voteFiltered, Err: makeSerErr(err)}
 		}
+		t.Freezer = newFreezer
```

Comment:
I think this looks good. Since both cases return filteredEvents and you're attaching the effect to both, I think this would basically give a more accurate error string.